docs/tutorials/sinks/2_http_sink.md

```diff
@@ -0,0 +1,504 @@
+        BasicRequest {
+            finalizers: metadata,
+            payload: payload.into_payload(),
+            metadata: request_metadata,
+        }
+    }
+```
+
+`build_request` is used to build the final request that will contain the encoded
+payload and the metadata. The `BasicRequest` object we return here is passed to 
+our `Tower` service where the data is actually sent.
+
+# Service
+
+We need to create a service that is responsible for actually sending our final
+encoded data.
+
+```rust
+struct BasicService {
+    endpoint: String,
+    client: HttpClient,
+}
+```
+
+The two fields the service contains, `endpoint` and `client` are the `endpoint`
+and `client` passed in from the `BasicSink` described earlier.
+
+`BasicService` implements the `tower::Service` trait:
+
+```rust
+impl tower::Service<BasicRequest> for BasicService {
+}
+```
+    
+A number of associated types need defining:
+
+```rust        
+    type Response = BasicResponse;
+    type Error = &'static str;
+    type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
+```
+    
+## poll_ready
+
+`poll_ready` is used to indicate when the service is ready to send data. This service
+has no reason to block, so we always return `Poll::Ready`.
+
+```rust
+    fn poll_ready(
+        &mut self,
+        _cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Result<(), Self::Error>> {
+        Poll::Ready(Ok(()))
+    }
+```
+
+## call
+
+`call` is where the data is actually sent over http. It returns a future that will
+be invoked to send the actual data.
+
+```rust
+    fn call(&mut self, request: BasicRequest) -> Self::Future {
+        let byte_size = request.payload.len();
+        let body = hyper::Body::from(request.payload);
+        let req = http::Request::post(&self.endpoint)
+            .header("Content-Type", "application/json")
+            .body(body)
+            .unwrap();
+
+        let mut client = self.client.clone();
+
+        Box::pin(async move {
+            match client.call(req).await {
+                Ok(response) => {
+                    if response.status().is_success() {
+                        Ok(BasicResponse { byte_size })
+                    } else {
+                        Err("received error response")
+                    }
+                }
+                Err(_error) => Err("oops"),
+            }
+        })
+    }
+```
+
+That future returns `BasicResponse`.
+
+# BasicResponse
+
+The return from our service must be an object that implements the `DriverResponse` trait.
+
+```rust
+struct BasicResponse {
+    byte_size: usize,
+}
+
+impl DriverResponse for BasicResponse {
+    fn event_status(&self) -> EventStatus {
+        EventStatus::Delivered
+    }
+
+    fn events_sent(&self) -> CountByteSize {
+        // (events count, byte size)
+        CountByteSize(1, self.byte_size)
+    }
+}
+```
+
+Vector calls the methods in this trait to determine if the event was delivered successfully.
```

Comment:
```suggestion
The `Tower` service infrastructure calls the methods in this trait to determine if the event was delivered successfully.
```

Reply:
These are called by the [`Driver`](https://github.com/vectordotdev/vector/blob/master/lib/vector-core/src/stream/driver.rs#L134-L187) in `vector-core`.